paasta_tools/smartstack_tools.py

```diff
@@ -412,42 +414,33 @@ def match_backends_and_tasks(
 _MesosSlaveDict = TypeVar('_MesosSlaveDict', bound=Dict)  # no type has been defined in mesos_tools for these yet.
 
 
+class SmartstackHost(NamedTuple):
+    hostname: str
+    pool: str
+
+
 class SmartstackReplicationChecker:
-    """Retrieves the number of registered instances in each discoverable location.
+    """Base class for checking smartstack replication. Extendable for different frameworks
 
     Optimized for multiple queries. Gets the list of backends from synapse-`roxy
     only once per location and reuse it in all subsequent calls of
     SmartstackReplicationChecker.get_replication_for_instance().
 
-    :Example:
-
-    >>> from paasta_tools.mesos_tools import get_slaves
-    >>> from paasta_tools.utils import load_system_paasta_config
-    >>> from paasta_tools.marathon_tools import load_marathon_service_config
-    >>> from paasta_tools.smartstack_tools import SmartstackReplicationChecker
-    >>>
-    >>> mesos_slaves = get_slaves()
-    >>> system_paasta_config = load_system_paasta_config()
-    >>> instance_config = load_marathon_service_config(service='fake_service',
-    ...                       instance='fake_instance', cluster='norcal-stagef')
-    >>>
-    >>> c = SmartstackReplicationChecker(mesos_slaves, system_paasta_config)
-    >>> c.get_replication_for_instance(instance_config)
-    {'uswest1-stagef': {'fake_service.fake_instance': 2}}
-    >>>
+    _get_allowed_locations_and_hosts must be implemented in super class
```

Comment:
s/super class/subclass/

Maybe use `abc.ABC` and decorate `_get_allowed_locations_and_hosts` with `@abc.abstractmethod`